internal/walker/rewriter_test.go

```diff
@@ -303,6 +303,69 @@ func TestRewriter(t *testing.T) {
 	}
 }
 
+func TestSnapshotRewriter(t *testing.T) {
+	tree := TestTree{
+		"foo": TestFile{Size: 21},
+		"bar": TestFile{Size: 21},
+		"subdir": TestTree{
+			"subfile": TestFile{Size: 21},
+		},
+	}
+	newTree := TestTree{
+		"foo": TestFile{Size: 42},
+		"subdir": TestTree{
+			"subfile": TestFile{Size: 42},
+		},
+	}
+	t.Run("", func(t *testing.T) {
+		repo, root := BuildTreeMap(tree)
+		expRepo, expRoot := BuildTreeMap(newTree)
+		modrepo := WritableTreeMap{repo}
+
+		ctx, cancel := context.WithCancel(context.TODO())
+		defer cancel()
+
+		rewriteNode := func(node *restic.Node, path string) *restic.Node {
+			if path == "/bar" {
+				return nil
+			}
+			if node.Type == "file" {
+				node.Size += 21
+			}
+			return node
+		}
+		rewriter, snapshotSizeRewriter := NewSnapshotSizeRewriter(rewriteNode)
+		newRoot, err := rewriter.RewriteTree(ctx, modrepo, "/", root)
+		if err != nil {
+			t.Error(err)
+		}
+
+		ss := snapshotSizeRewriter()
+
+		if ss.FileCount != 2 {
```

Comment:
Please replace the if-block with `test.Equals(t, uint(2), ss.FileCount, "snapshot file count mismatch")`. (Similarly for `FileSize` below).